pulser/sequence.py

```diff
@@ -296,11 +319,12 @@ def declare_channel(self, name, channel_id, initial_target=None):
 
         # Manually store the channel declaration as a regular call
         self._calls.append(_Call(
-                            "declare_channel",
-                            (name, channel_id),
-                            {"initial_target": initial_target}))
+            "declare_channel",
+            (name, channel_id),
+            {"initial_target": initial_target}))
 
-    def declare_variable(self, name, size=1, dtype=float):
+    def declare_variable(self, name: str, size: int = 1,
+                         dtype: type = float) -> Variable:
```

Comment:
Let's specify the possible types here: `dtype` can be `Union[type[int], type[float], type[str]]`